apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/defaultimpl/DefaultEmailService.java

```diff
@@ -1,12 +1,102 @@
 package com.ctrip.framework.apollo.portal.spi.defaultimpl;
 
+import com.ctrip.framework.apollo.portal.component.config.PortalConfig;
 import com.ctrip.framework.apollo.portal.entity.bo.Email;
 import com.ctrip.framework.apollo.portal.spi.EmailService;
+import com.ctrip.framework.apollo.tracer.Tracer;
+import com.sun.mail.smtp.SMTPTransport;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Properties;
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.annotation.Resource;
+import javax.mail.Message;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public class DefaultEmailService implements EmailService{
+public class DefaultEmailService implements EmailService {
+
+  private final Logger logger = LoggerFactory.getLogger(DefaultEmailService.class);
+
+  @Resource
+  private PortalConfig portalConfig;
 
   @Override
-  public void send(Email email){
-    //do nothing
+  public void send(Email email) {
+    if (!portalConfig.isEmailEnabled()) {
+      return;
+    }
+
+    SMTPTransport t = null;
+    try {
+      Properties prop = System.getProperties();
+      prop.put("mail.smtp.auth", "true");
```

Comment:
Is this mandatory? I noticed the following instruction from the [javadoc](https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html)

> To use SMTP authentication you'll need to set the mail.smtp.auth property (see below) or provide the SMTP Transport with a username and password when connecting to the SMTP server.

Reply:
Thanks, I just remove this line. The test is fine. Please take a look.